util/dvsim/FlowCfg.py

```diff
@@ -243,11 +245,18 @@ def resolve_hjson_raw(self, hjson_dict):
                     cfg_file = subst_wildcards(entry,
                                                self.__dict__,
                                                ignore_error=True)
+                    # skip sub-cfgs that are not in the list
```

Comment:
Prefer to not change the `resolve_hjson_raw` function since it is already quite complicated. 

Instead, add `prune_selected_cfgs()` and call it at the start of `create_deploy_objects()`. 

`prune_selected_cfgs()` will return immediately if not a master cfg list or if `--select-cfgs` is None. If not, then it will go through the list of `self.cfgs` and remove the ones not selected via `--select-cfgs` switch. This IMO is a lot easier to implement.  

This way, in all other functions that loop over `self.cfgs`, it will only loop over the cfgs selected on the cmd line.

Reply:
I see, thanks Sri for the detailed suggestion. I was removing them in `parse_flow_cfg` because in `SimCfg.py` function `__init__`, we are checking the modes for all self.cfgs and printing out the scratch directories.
But I think your idea is cleaner. I will try to clean up the `__init__` function. Thanks!